### PR TITLE
Fix crash in completion with bad TypeConstraints

### PR DESCRIPTION
**Pull request by jez**
### Motivation

This was causing lots of crashes in LSP mode.

### Test plan

I confirmed that the included test caused a crash before this change.

The exception happened in `TypeConstraint::findSolution` because we would be
looking in a solved type constraint for the solution to a type variable that did
not exist in that `TypeConstraint`'s solution set.

### CODE REVIEW

File: test/testdata/lsp/completion/constr.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# typed: true
+
+extend T::Sig
+
+xs = T.let([], T::Array[Integer])
+
+# This completion result is a bit non-sensical at the moment (it doesn't makes
+# sense to return any results there) but at least it doesn't crash Sorbet like
+# it used to. If you made a change that improves this test, feel free to delete
+# this comment.
```
Comment (elliottt) [inline test/testdata/lsp/completion/constr.rb:10]: What ends up being nonsensical about the completion result? The assertion on line 15 is for `each`, which doesn't seem crazy for a tab completion on the result of a `map`.

File: main/lsp/requests/completion.cc [hunk 2]
```diff
@@ -912,9 +906,18 @@ vector<unique_ptr<CompletionItem>> CompletionTask::getCompletionItems(LSPTypeche
...
     }
     for (auto &similarMethod : dedupedSimilarMethods) {
+        // Even though we might have one or more TypeConstraints on the DispatchResult that triggered this completion
+        // request, those constraints are the result of solving the current method. These new methods we're about to
+        // suggest are their own methods with their own type variables, so it doesn't make sense to use the old
+        // constraint for the new methods.
+        //
+        // What this means in practice is that the prettified `sig` in the completion documentation will show
+        // `T.type_parameter(:U)` instead of a solved type.
+        auto constr = nullptr;
```
Comment (elliottt) [inline main/lsp/requests/completion.cc:916]: Thanks for this comment, it really cleared up the deletions above 👍 

Sort of amazing that `nullptr` works here, is the deduced type `std::nullptr_t`?

**Review by elliottt — APPROVED** [review]
Looks good 👍

File: test/testdata/lsp/completion/constr.rb (hunk 1 above)
Comment (jez) [reply test/testdata/lsp/completion/constr.rb:10]: arguably that would make sense if you asked for completion results at the end of the `each` method name, not after the `end` keyword. IMO it's kind of like asking for completion results right after a `}`: we should probably show you other results or no results. you're right that it's subjective—i was just trying to let people know that i'm not attached to the current completion results.

File: main/lsp/requests/completion.cc (hunk 2 above)
Comment (jez) [reply main/lsp/requests/completion.cc:916]: > Sort of amazing that nullptr works here, is the deduced type std::nullptr_t?

yep

**Pull request merged**